test/cmocka/src/math/arithmetic/base_10_logarithm.c

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <string.h>
+
+#include <sof/math/base2log.h>
+#include <sof/audio/format.h>
+#include <sof/string.h>
+#include <sof/common.h>
+#include "log2_tables.h"
+/* 'Error[max] = 0.0000122948805110,THD = -98.2055137527489421 ' */
+#define CMP_TOLERANCE 0.0000122948805111 /*0.0001079666317204*/
+/* Natural logarithm log10(X) reference table generated by matlab/Octave */
```

Comment:
Why not just use `<math.h>`'s log functions, why is this table needed?

Reply:
Using book and pen log function calculation i.e. manual calculation. In  most of the case, Matlab/Octave log function calculation is more accurate than built-in `<math.h>`, library functions.

Reply:
Log table is a reference table which is derived from offline log function using `Octave/matlab`. Reference table values are compared against fix point function.

Reply:
@marc-hb we dont use C lib maths as they are not optimised for xtensa or SOF formats.

Reply:
This seems vulnerable to a copy/paste typo. Is there some redundancy built in in the test? If not, then adding another test, very simple test using `<math.h>` (when available) would provide that redundancy.